### PR TITLE
Make audit models str method python 2 and 3 compatible

### PR DESCRIPTION
**Pull request by caktus-philip**
Ensure all models in audit have Python 3-compatible `str()` methods using Django's `@python_2_unicode_compatible` decorator. Fixed a bug in `AuditorStaffMember`.

https://docs.djangoproject.com/en/1.11/ref/utils/#django.utils.encoding.python_2_unicode_compatible

Add non-ASCII tests for `str()` and `unicode()` methods for most audit models. 

In `EquiTrack/tests/test_all_models.py`, remove `audit` from the list of modules not tested for Python 3-compatibility since it's now compliant.

### CODE REVIEW

**Comment by reinbach** [comment]
How come we're merging into `prp-refactoring`, should I be changing my pull requests to merge with that branch as well?

File: EquiTrack/audit/tests/test_models.py [hunk 1]
```diff
@@ -1,8 +1,24 @@
+from __future__ import absolute_import
```
Comment (reinbach) [inline EquiTrack/audit/tests/test_models.py]: Looks like `audit/models.py` is lacking all these `__future__` imports

File: EquiTrack/audit/tests/test_models.py [hunk 2]
```diff
@@ -21,3 +37,93 @@ def test_signal(self):
         self.assertIn(Auditor.name, staff_member.user.groups.values_list('name', flat=True))
 
         self.assertEqual(len(mail.outbox), 1)
+
+
+@skipIf(sys.version_info.major == 3, "This test can be deleted under Python 3")
```
Comment (reinbach) [inline EquiTrack/audit/tests/test_models.py]: Nice!

**Comment by ntrncic** [comment]
this one can be against develop since it doesn't have migrations and doesn't touch partner and report models

File: EquiTrack/audit/tests/test_models.py (hunk 1 above)
Comment (caktus-philip) [reply EquiTrack/audit/tests/test_models.py]: Thanks @reinbach. Earlier this year we (Vinod & I) wrote recommendations for UNICEF and the recommendation was to add the 4  `from __future__` imports to all *new* files. But since adding them to existing code can can change behavior, we recommend leaving existing code as-is unless adding a `from __future__` declaration solves a specific problem.

Comment (reinbach) [reply EquiTrack/audit/tests/test_models.py]: :+1: thanks for the clarifcation

Comment (caktus-philip) [reply EquiTrack/audit/tests/test_models.py]: But I guess I added all 4 `from __future__` imports to the existing file `test_models.py`, so not following my own recommendation. 

` ¯\_(ツ)_/¯`

**Comment by caktus-philip** [comment]
Replaced by #993
